SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h

```diff
@@ -70,29 +70,55 @@ class SPATIALGDK_API UEventTracingSamplingSettings : public UObject
 {
 	GENERATED_BODY()
 public:
+	struct TraceQueryDeleter
+	{
+		void operator()(Trace_Query* Query) const
+		{
+			if (Query != nullptr)
+			{
+				Trace_Query_Destroy(Query);
+			}
+		}
+	};
+
+	using TraceQueryPtr = TUniquePtr<Trace_Query, TraceQueryDeleter>;
+
 	UPROPERTY(EditAnywhere, Category = "Event Tracing", meta = (ClampMin = 0.0f, ClampMax = 1.0f))
 	double SamplingProbability = 1.0f;
 
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
 	TMap<FName, double> EventSamplingModeOverrides;
 
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
-	FString GDKEventPreFilter;
+	FString GDKEventPreFilter = "false";
 
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
-	FString GDKEventPostFilter;
+	FString GDKEventPostFilter = "false";
 
 	/* The runtime filter which is used for local/cloud editor workflows (generated configs). */
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
-	FString RuntimeEventPreFilter;
+	FString RuntimeEventPreFilter = "false";
 
 	/* The runtime filter which is used for local/cloud editor workflows (generated configs). */
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
-	FString RuntimeEventPostFilter;
+	FString RuntimeEventPostFilter = "false";
+
+	FString GetGDKEventPreFilterString() const { return IsFilterValid(GDKEventPreFilter) ? GDKEventPreFilter : "false"; }
```

Comment:
Minor: I would suggest generalising this into a GetFilterString function that takes a string parameter.
```
static const FString NoneFilter = "None";
FString GetPreFilterString(const FString& EventType) const 
{
	return IsFilterValid(EventType) ? EventType : NoneFilter;
}
FString GetGDKEventPreFilterString() const { return GetPreFilterString(GDKEventPreFilter); }
// etc
```